src/normalize/normalize.rs

```diff
@@ -38,6 +43,52 @@ pub struct NormalizedAddrs<M> {
 pub type NormalizedUserAddrs = NormalizedAddrs<UserAddrMeta>;
 
 
+/// A type representing a build ID note.
+///
+/// In the ELF file, this header is typically followed by the variable sized
+/// build ID.
+#[repr(C)]
+struct BuildIdNote {
+    /// ELF note header.
+    header: Elf64_Nhdr,
+    /// NUL terminated string representing the name.
+    name: [c_char; 4],
+}
+
+// SAFETY: `BuildIdNote` is valid for any bit pattern.
+unsafe impl crate::util::Pod for BuildIdNote {}
+
+
+/// Attempt to read an ELF binary's build ID.
+// TODO: Currently look up is always performed based on section name, but there
+//       is also the possibility of iterating notes and checking checking
+//       Elf64_Nhdr.n_type for NT_GNU_BUILD_ID, specifically.
+fn read_build_id(path: &Path) -> Result<Option<String>> {
+    let file = File::open(path)?;
+    let parser = ElfParser::open_file(file)?;
+
+    // The build ID is contained in the `.note.gnu.build-id` section. See
+    // elf(5).
+    if let Ok(idx) = parser.find_section(".note.gnu.build-id") {
```

Comment:
let's at least do sanity checking of note type and name ("GNU")? As you mention in TODO above, kernel and BPF selftests implementations (https://lore.kernel.org/all/20230331093157.1749137-3-jolsa@kernel.org/ and https://github.com/torvalds/linux/blob/master/lib/buildid.c)  don't make an assumption of ".note.gnu.build-id" section (I've been told it's not guaranteed/specified by any starndard, so best not to make assumptions), so maybe let's do it more robustly from the very beginning?